src/shared/file_op.c

```diff
@@ -754,7 +754,9 @@ int UnmergeFiles(const char *finalpath, const char *optdir, int mode, const char
         }
 
         /* Mv to original name */
-        rename_ex(tmp_file, final_name);
+        if (rename_ex(tmp_file, final_name) != 0) {
+            ret = 0;
```

Comment:
Wouldn't it seem appropriate to clean up and go out? I mean that this code would let the code fill `unmerged_files` and also leave the temporary file missing.

```suggestion
            unlink(tmp_file);
            ret = 0;
            break;
```